src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,144 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        tail = newNode;
+        if (size == 0) {
+            head = newNode;
+        } else {
+            tail.prev.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index <= size && index >= 0) {
```

Comment:
You can use `checkIfIndexIsValid` method here instead of `if else`

Reply:
I can't use here my checkIfIndexIsValid method, because in the metho body such condition if (!(index < size && index >= 0)). But create new method for only one condition it's no sense.